records/2022-04-14.secrets-usage.md

```diff
@@ -109,37 +109,37 @@ that are not available through the usual API offered to Notify users.
+| Inbound SMS                                                                | Yes        | No             | The inbound SMS content is signed and stored into the database.                                                                 | 1. [Signing SMS content prior to storage](https://github.com/cds-snc/notification-api/blob/9461b26f9582bdea6fe7c68b3d360597e5eab9ba/app/models.py#L2118)                                                                                                                                                                                                                                                                                                                                                                                                                                    |   |
+| Storing notification in Redis / SQS queues                                 | Yes        | No             | The complete notification is signed and stored into the queue.                                                                  | 1. [Signing notification content prior to storage](https://github.com/cds-snc/notification-api/blob/02d431774ced0e2bc1dbe19ee8ceddcd1a8371dc/app/celery/tasks.py#L159-171)                                                                                                                                                                                                                                                                                                                                                                                                                  |   |
+| Password reset code / Verification URL / Confirmation URL / Invitation URL | Yes        | Yes            | Expires after 1 hour.                                                                                                           | 1. [url_with_tokens](https://github.com/cds-snc/notification-api/blob/bdf1174440272f3bc6d05587cb9100edbe12e3a9/app/utils.py#L29-L34)                                                                                                                                                                                                                                                                                                                                                                                                                                                        |   |
```

Comment:
These URLs still use the salt per he code review I have done. @sastels I assume you might prefer a contextual salt for these as well, right?

Reply:
So the app uses the utils repo to generate these. I've added code to utils to use the salt "token" as a default but currently api is still sending in DANGEROUS_SALT. Working on a slight tweak to utils today to make "token" used for creating the token and (either) "token" or DANGEROUS_SALT used for verifying. Once this is merged and the codebase is using it for a day, all the old DANGEROUS_SALT tokens will have expired and we can switch to verifying with only "token".

So TL;DR What you have here is correct and can be merged in. We can update next week when we're not using DANGEROUS_SALT at all for the tokens.